meshtastic/interfaces/ble/reconnection.py

```
@@ -0,0 +1,257 @@
+        reconnect policy stops further retries, the provided shutdown_event is set, or auto_reconnect
+        is False. Between failed attempts the loop respects the policy's backoff delay (adjusted for
+        specific BLE/DBus errors) and allows early exit when shutdown_event is signaled. On exit the
+        scheduler's thread reference is cleared.
+        
+        Parameters:
+            auto_reconnect (bool): If False, exit immediately without attempting reconnects.
+            shutdown_event (threading.Event): Event that, when set, causes the loop to stop as soon as possible.
+        """
+        self.reconnect_policy.reset()
+
+        sleep_fn = get_sleep_fn()
```

Ruff (0.15.0)

[error] Local variable `sleep_fn` is assigned to but never used

Remove assignment to unused variable `sleep_fn`

(F841)